### PR TITLE
[aes, pre_dv] Align S-Box Verilator TB with latest changes to DOM S-Box

### PR DESCRIPTION
**Pull request by vogelpi**
This change is ensuring that the internally buffered entropy is updated such that the next S-Box evaluation will get fresh entropy for the first multiplier stage.

This is related to https://github.com/lowRISC/opentitan/pull/18480

### CODE REVIEW

**Comment by vogelpi** [comment]
The SV file touched by this PR is a pre_dv testbench file. This is needed to re-verify the masking using formal on the production netlist. This PR doesn't affect the real DV nor the RTL.

CHANGE AUTHORIZED: hw/ip/aes/pre_dv/aes_sbox_tb/rtl/aes_sbox_tb.sv

**Comment by GregAC** [comment]
CHANGE AUTHORIZED: hw/ip/aes/pre_dv/aes_sbox_tb/rtl/aes_sbox_tb.sv

**Comment by GregAC** [comment]
Authorized the change as I'm happy this doesn't effect RTL but not actually reviewing, I'll leave that it @ballifatih

**Review by ballifatih — APPROVED** [review]
LGTM, thanks @vogelpi. I'll approve but I think you also need approval from a committer of the repo.

File: hw/ip/aes/pre_dv/aes_sbox_tb/rtl/aes_sbox_tb.sv
Comment (andreaskurth) [inline hw/ip/aes/pre_dv/aes_sbox_tb/rtl/aes_sbox_tb.sv:119]: This isn't equivalent to how the RTL gets changed in 1d1434d2d1ec773cb2bf6d2f2239e35de40461b5, right? If so, should we add a comment explaining why we can simply use `out_req_o` to drive `prd_we_i` in this TB but don't do that in a real AES cipher core?

Comment (vogelpi) [reply hw/ip/aes/pre_dv/aes_sbox_tb/rtl/aes_sbox_tb.sv]: It actually is inline with the RTL change. The internally buffered PRD such that it's presented to the first multiplier stage in sync with the new input data. Anyway, for this testbench it doesn't really matter because it's just about functional verification. I've added a comment to explain this. PTAL.

**Pull request merged**
